From 330564fff5946c0cd11b6744443bae95cf54019b Mon Sep 17 00:00:00 2001
From: Emilio Jesus Gallego Arias <e+git@x80.org>
Date: Tue, 30 Jan 2018 21:10:58 +0100
Subject: [PATCH] [make] Fix build in split `num` configuration.

It seems that recent fixes using ocamlbuild to locate num missed to
switch in some binaries. This is needed to solve problems like the
one in this issue: https://github.com/ocaml/opam-repository/issues/11316
---
 Makefile.build    | 2 +-
 tools/coqmktop.ml | 3 ++-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/Makefile.build b/Makefile.build
index b0e7546113d..312ce53dd88 100644
--- a/Makefile.build
+++ b/Makefile.build
@@ -530,7 +530,7 @@ CSDPCERTCMO:=lib/clib.cma $(addprefix plugins/micromega/, \
 
 $(CSDPCERT): $(call bestobj, $(CSDPCERTCMO))
 	$(SHOW)'OCAMLBEST -o $@'
-	$(HIDE)$(call bestocaml,,nums unix)
+	$(HIDE)$(call bestocaml,-linkpkg -package num -package unix,)
 
 ###########################################################################
 # tests
diff --git a/tools/coqmktop.ml b/tools/coqmktop.ml
index 950ed53ccf7..c387c879824 100644
--- a/tools/coqmktop.ml
+++ b/tools/coqmktop.ml
@@ -108,7 +108,7 @@ let incl_all_subdirs dir opts =
 
 (** OCaml + CamlpX libraries *)
 
-let ocaml_libs = ["str.cma";"unix.cma";"nums.cma";"dynlink.cma";"threads.cma"]
+let ocaml_libs = ["str.cma";"dynlink.cma"]
 let camlp4_libs = ["gramlib.cma"]
 let libobjs = ocaml_libs @ camlp4_libs
 
@@ -289,6 +289,7 @@ let main () =
       List.filter ((<>) "") (split_on_char ' ' Coq_config.caml_flags) in
     let args =
       coq_camlflags @ "-linkall" :: "-w" :: "-31" :: flags @ copts @ options @
+      ["-linkpkg"; "-package"; "num"] @
       (std_includes basedir) @ tolink @ [ main_file ] @ topstart
     in
     if !echo then begin
